.github/workflows/scheduled.yml

```diff
@@ -315,3 +323,44 @@ jobs:
           name: join-fuzzer-failure-artifacts
           path: |
             /tmp/join_fuzzer_repro
+
+  linux-exchange-fuzzer-run:
+    runs-on: ubuntu-latest
+    needs: compile
+    timeout-minutes: 120
+    steps:
+
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          ref: "${{ inputs.ref || 'main' }}"
+
+      - name: Install dependencies
+        run: source ./scripts/setup-ubuntu.sh
+
+      - name: Download exchange fuzzer
+        uses: actions/download-artifact@v3
+        with:
+          name: exchange
+
+      - name: Run exchange Fuzzer
+        run: |
+          mkdir -p /tmp/exchange_fuzzer_repro/
+          rm -rfv /tmp/exchange_fuzzer_repro/*
+          chmod -R 777 /tmp/exchange_fuzzer_repro
+          chmod +x velox_exchange_fuzzer_test
+          ./velox_exchange_fuzzer_test \
+                --seed ${RANDOM} \
+                --duration_sec 1800 \
+                --logtostderr=1 \
+                --minloglevel=0 \
+                --repro_path=/tmp/exchange_fuzzer_repro
+            && echo -e "\n\Exchange fuzzer run finished successfully."
+
+      - name: Archive Exchange production artifacts
+        if: always()
```

Comment:
Why do we need this statement? Isn't it always true?

Reply:
Yes, because typically the artifact mechanism is used when the job produces some artifact at the end of the run (think you are building a package etc) that you want to upload . When the job fails most jobs dont want to upload half built packages etc, however in our case its the opposite as when we fail we want to save the repro and thus the always flag.

Reply:
Got it. Thank you for explaining.